extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-a4a-config.js

```diff
@@ -88,17 +88,37 @@ describe('doubleclick-a4a-config', () => {
     });
 
     it('should select into canonical AMP experiment when not on CDN', () => {
-      forceExperimentBranch(mockWin, DFP_CANONICAL_FF_EXPERIMENT_NAME,
-          DOUBLECLICK_EXPERIMENT_FEATURE.CANONICAL_EXPERIMENT);
       sandbox.stub(DoubleclickA4aEligibility.prototype,
           'isCdnProxy', () => false);
+      const maybeSelectExperimentSpy = sandbox.spy(
+          DoubleclickA4aEligibility.prototype, 'maybeSelectExperiment');
       const elem = testFixture.doc.createElement('div');
       testFixture.doc.body.appendChild(elem);
-      expect(doubleclickIsA4AEnabled(mockWin, elem)).to.be.true;
-      expect(elem.getAttribute(EXPERIMENT_ATTRIBUTE)).to.equal(
-          DOUBLECLICK_EXPERIMENT_FEATURE.CANONICAL_EXPERIMENT);
+      doubleclickIsA4AEnabled(mockWin, elem);
+      expect(maybeSelectExperimentSpy).to.be.calledWith(mockWin, elem, [
+        DOUBLECLICK_EXPERIMENT_FEATURE.CANONICAL_CONTROL,
+        DOUBLECLICK_EXPERIMENT_FEATURE.CANONICAL_EXPERIMENT,
+      ], DFP_CANONICAL_FF_EXPERIMENT_NAME) ;
+
     });
 
+    it('should select into non-SSL canonical AMP experiment when not on CDN',
+        () => {
+          sandbox.stub(DoubleclickA4aEligibility.prototype,
+              'isCdnProxy', () => false);
+          sandbox.stub(DoubleclickA4aEligibility.prototype,
+              'supportsCrypto', () => false);
+          const maybeSelectExperimentSpy = sandbox.spy(
+              DoubleclickA4aEligibility.prototype, 'maybeSelectExperiment');
+          const elem = testFixture.doc.createElement('div');
+          testFixture.doc.body.appendChild(elem);
+          doubleclickIsA4AEnabled(mockWin, elem);
+          expect(maybeSelectExperimentSpy).to.be.calledWith(mockWin, elem, [
```

Comment:
To fully ensure proper behavior can you add:  expect(maybeSelectExperimentSpy.callCount).to.equal(1);  Want to make sure it was not called again...

Reply:
done